nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,677 @@
+            self.cv_y_true_.append(y_true)
+            self.cv_y_pred_.append(y_pred)
+            self.cv_scores_.append(score_func(y_true, y_pred))
+
+        self.coef_ = np.vstack([np.mean(coefs[c], axis=0) for c in classes])
+        self.std_coef_ = np.vstack([np.std(coefs[c], axis=0) for c in classes])
+        self.intercept_ = np.hstack([np.mean(intercepts[c], axis=0)
+                                     for c in classes])
+        coef_img = {}
+        std_coef_img = {}
+        for c, coef, std in zip(classes, self.coef_, self.std_coef_):
+            c = 'beta' if c is None else c
+            coef_img[c] = self.masker_.inverse_transform(coef)
+            std_coef_img[c] = self.masker_.inverse_transform(std)
+
+        self.coef_img_ = coef_img
+        self.std_coef_img_ = std_coef_img
+        return self
+
+    def fit(self, niimgs, y, index=None):
+        """Fit the decoder
+
+        Parameters
+        ----------
+        niimgs: list of filenames or NiImages
+            Data on which the mask must be calculated. If this is a list,
+            the affine is considered the same for all.
+
+        y: 1D array-like
+           Target variable to predict. Must have exactly as many elements as
+           3D images in niimg.
+
+        index: Any type compatible with numpy array indexing
+            Used for indexing the 4D NiImages in the fourth dimension.
+
+        Attributes
+        ----------
+        `mask_img_`: NiImage
+            Mask computed by the masker object.
+        `classes_`: numpy.ndarray
+            Classes to predict. For classification only.
+        `coef_`: numpy.ndarray, shape=(n_classes, n_features)
+            Contains the mean of the models weight vector across
+            fold for each class.
+        `std_coef_`: numpy.ndarray, shape=(n_classes, n_features)
+            Contains the standard deviation of the models weight vector across
+            fold for each class.
+        `coef_img_`: dict of NiImage
+            Dictionary containing `coef_` with class names as keys,
+            and `coef_` transformed in NiImages as values. In the case
+            of a regression, it contains a single NiImage at the key 'beta'.
+        `std_coef_img_`: dict of NiImage
+            Dictionary containing `std_coef_` with class names as keys,
+            and `coef_` transformed in NiImages as values. In the case
+            of a regression, it contains a single NiImage at the key 'beta'.
+        `cv_y_true_` : numpy.ndarray
+            Ground truth labels for left out samples in inner cross validation.
+        `cv_y_pred_` : numpy.ndarray
+            Predicted labels for left out samples in inner cross validation.
+        `cv_params_`: dict of lists
+            Best point in the parameter grid for each tested fold
+            in the inner cross validation loop.
+        """
+        # Setup memory, parallel and masker
+        if isinstance(self.memory, basestring) or self.memory is None:
```

Comment:
If I am not wrong. You can validate basestring using from nilearn. ```nilearn._utils.compat```. I think this might be reason for circleci failure.